### PR TITLE
halion-sonic 7.1.20,79d9ffab-0816-4c13-b3df-d21a52ddbe48 (new cask)

### PR DESCRIPTION
**Pull request by zyoshoka**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/h/halion-sonic.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+cask "halion-sonic" do
+  version "7.1.20,79d9ffab-0816-4c13-b3df-d21a52ddbe48"
+  sha256 "956b3bafa767f5d15d2818f59d10c873a8db087d515ff970ab0708cbdc05f224"
+
+  url "https://download.steinberg.net/automated_updates/sda_downloads/#{version.csv.second}/Halion_Sonic_#{version.csv.first}_Installer_mac.dmg"
+  name "HALion Sonic #{version.csv.first.major}"
```
Comment (khipp) [inline Casks/h/halion-sonic.rb:6]:
```suggestion
  name "HALion Sonic"
```

File: Casks/h/halion-sonic.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  desc "Player for sample libraries, synthesizers and hybrid instruments"
+  homepage "https://o.steinberg.net/en/support/downloads/halion_sonic_#{version.csv.first.major}.html"
```
Comment (khipp) [inline Casks/h/halion-sonic.rb:8]:
```suggestion
  homepage "https://www.steinberg.net/vst-instruments/halion/sonic/"
```

File: Casks/h/halion-sonic.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  livecheck do
+    url :homepage
+    regex(%r{href=.*/sda_downloads/([^/]+)/Halion[._-]Sonic[._-]v?(\d+(?:\.\d+)+)[._-]Installer[._-]mac\.dmg}i)
```
Comment (khipp) [inline Casks/h/halion-sonic.rb:12]:
```suggestion
    url "https://o.steinberg.net/en/support/downloads/halion_sonic_#{version.csv.first.major}.html"
    regex(%r{href=.*?downloads/([^/]+)/Halion[._-]Sonic[._-]v?(\d+(?:\.\d+)+)[._-]Installer[._-]mac\.dmg}i)
```

**Review by khipp — APPROVED** [review]

File: Casks/h/halion-sonic.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+      "#{match[2]},#{match[1]}"
+    end
+  end
+
+  depends_on cask: [
+    "steinberg-activation-manager",
+    "steinberg-library-manager",
+    "steinberg-mediabay",
+  ]
```
Comment (bevanjkay) [inline Casks/h/halion-sonic.rb:25]: Can you explain the nature of the dependencies here?
I tend to err on the side of caution with forcing other apps to be installed through `homebrew-cask`, especially when they may be shared with other applications that aren't installed this way. Running `brew uninstall halion-sonic` may remove the dependencies if other apps also use these but aren't installed with Homebrew.

Comment (zyoshoka) [reply Casks/h/halion-sonic.rb:25]: The dependencies are described on https://o.steinberg.net/en/support/downloads/halion_sonic_7.html, and I have written `depends_on` stanza similarly in #194230. Should I leave this URL as a comment?

**Pull request merged**